ale_python_interface/ale_c_wrapper.cpp

```diff
@@ -1,3 +1,20 @@
-//all code is currently in the .h file
 #include "ale_c_wrapper.h"
 
+#include <cstring>
+#include <string>
+
+void encodeState(ALEState *state, char *buf) {
+	std::string str = state->serialize();
+
+	memcpy(buf, str.data(), str.length());
```

Comment:
Because that's the C wrapper, and C convention is generally to pass in a target buf. It wouldn't work with FFI if you had to pass in a vector.